skopt/optimizer/optimizer.py

```diff
@@ -176,7 +186,101 @@ def _check_arguments(self, base_estimator, n_random_starts, acq_optimizer):
                              "'sampling', got {0}".format(acq_optimizer))
         self.acq_optimizer = acq_optimizer
 
-    def ask(self):
+    def copy(self, copy_random_state=True):
+        """Creates a shallow copy of optimizer."""
+
+        params = {
+            'dimensions':self.space.dimensions,
+            'base_estimator':self.base_estimator,
+            'n_random_starts':self._n_random_starts,
+            'acq_func':self.acq_func,
+            'acq_optimizer':self.acq_optimizer,
+            'acq_func_kwargs':self.acq_func_kwargs,
+            'acq_optimizer_kwargs':self.acq_optimizer_kwargs,
+        }
+
+        if copy_random_state:
+            params['random_state'] = copy.copy(self.rng)
+
+        optimizer = Optimizer(**params)
+
+        if hasattr(self, "gains_"):
+            optimizer.gains_ = np.copy(self.gains_)
+
+        if len(self.Xi) > 0:
+            optimizer.tell(self.Xi, self.yi)
+
+        return optimizer
+
+    def ask(self, n_points=None, strategy="cl_min"):
+        """Query point or multiple points at which objective should be evaluated.
+
+        * `n_points` [int or None]:
+            Number of points returned by the ask method.
+            If the value is None, a single point to evaluate is returned.
+            Otherwise a list of points to evaluate is returned of size n_points.
+            This is useful if you can evaluate your objective in parallel,
+            and thus obtain more objective function evaluations per unit of time.
+
+        * `strategy` [string, default=`"cl_min"`]:
+            Method to use to sample multiple points (see also `n_points` description).
+            This parameter is ignored if n_points = None. Supported options are `"cl_min"`,
+            `"cl_mean"` or `"cl_max"`.
+
+            - If set to `"cl_min"`, then constant liar strtategy is used
+               with lie objective value being minimum of observed objective values.
+               `"cl_mean"` and `"cl_max"` means mean and max of values respectively.
+               For details on this strategy see:
+
+               https://hal.archives-ouvertes.fr/hal-00732512/document
+
+               With this strategy a copy of optimizer is created, which is then asked
+               for a point, and the point is told to the copy of optimizer with
+               some fake objective (lie), the next point is asked from copy,
+               it is also told to the copy with fake objective and so on.
+               The type of lie defines different flavours of `cl_x` strategies.
+
+        """
+        if n_points is None:
```

Comment:
What should the right behaviour be if `n_points == 1`? I think it should return a list with just one value, which makes it different from `None`. I think we agree but wanted to double check